backend/controller/dal/dal_test.go

```diff
@@ -400,3 +401,93 @@ func assertEventsEqual(t *testing.T, expected, actual []Event) {
 	t.Helper()
 	assert.Equal(t, normaliseEvents(expected), normaliseEvents(actual))
 }
+
+func TestModuleConfiguration(t *testing.T) {
+	ctx := log.ContextWithNewDefaultLogger(context.Background())
+	conn := sqltest.OpenForTesting(ctx, t)
+	dal, err := New(ctx, conn)
+	assert.NoError(t, err)
+	assert.NotZero(t, dal)
+
+	tests := []struct {
+		TestName     string
+		ModuleSet    optional.Option[string]
+		ModuleGet    optional.Option[string]
+		PresetGlobal bool
+	}{
+		{
+			"SetModuleGetModule",
+			optional.Some("echo"),
+			optional.Some("echo"),
+			false,
+		},
+		{
+			"SetGlobalGetGlobal",
+			optional.None[string](),
+			optional.None[string](),
+			false,
+		},
+		{
+			"SetGlobalGetModule",
+			optional.None[string](),
+			optional.Some("echo"),
+			false,
+		},
+		{
+			"SetModuleOverridesGlobal",
+			optional.Some("echo"),
+			optional.Some("echo"),
+			true,
+		},
+	}
+
+	b := []byte(`"asdf"`)
+	for _, test := range tests {
+		t.Run(test.TestName, func(t *testing.T) {
+			if test.PresetGlobal {
+				err := dal.db.SetModuleConfiguration(ctx, optional.None[string](), "configname", []byte(`"qwerty"`))
+				assert.NoError(t, err)
+			}
+			err := dal.db.SetModuleConfiguration(ctx, test.ModuleSet, "configname", b)
+			assert.NoError(t, err)
+			gotBytes, err := dal.db.GetModuleConfiguration(ctx, test.ModuleGet, "configname")
+			assert.NoError(t, err)
+			assert.Equal(t, b, gotBytes)
+			err = dal.db.UnsetModuleConfiguration(ctx, test.ModuleGet, "configname")
+			assert.NoError(t, err)
+		})
+	}
+
+	t.Run("List", func(t *testing.T) {
+		sortedList := []sql.ModuleConfiguration{
+			{
+				Module: optional.Some("echo"),
+				Name:   "a",
+			},
+			{
+				Module: optional.Some("echo"),
+				Name:   "b",
+			},
+			{
+				Module: optional.None[string](),
+				Name:   "a",
+			},
+		}
+
+		// Insert entries in a separate order from how they should be returned to
+		// test sorting logic in the SQL query
+		err := dal.db.SetModuleConfiguration(ctx, sortedList[1].Module, sortedList[1].Name, []byte(`""`))
+		assert.NoError(t, err)
+		err = dal.db.SetModuleConfiguration(ctx, sortedList[2].Module, sortedList[2].Name, []byte(`""`))
+		assert.NoError(t, err)
+		err = dal.db.SetModuleConfiguration(ctx, sortedList[0].Module, sortedList[0].Name, []byte(`""`))
+		assert.NoError(t, err)
```

Comment:
It looks like we're still using the `dal.db` here instead of a method on the `dal` itself. If we go straight to the db, I think it will bypass the usefulness of the dal. You might want to duplicate the pattern here: https://github.com/TBD54566975/ftl/blob/main/backend/controller/cronjobs/cronjobs.go#L69 which calls a `dal` method here: https://github.com/TBD54566975/ftl/blob/main/backend/controller/dal/dal.go#L990

Reply:
Ah, thanks for catching that! I'll make that change now.